src/main/java/picard/arrays/illumina/CompareGtcFiles.java

```diff
@@ -0,0 +1,236 @@
+
+        try {
+            InfiniumNormalizationManifest infiniumNormalizationManifest
+                    = new InfiniumNormalizationManifest(ILLUMINA_NORMALIZATION_MANIFEST);
+            InfiniumGTCFile gtcFileOne = new InfiniumGTCFile(new DataInputStream(new FileInputStream(INPUT.get(0))), infiniumNormalizationManifest);
+            InfiniumGTCFile gtcFileTwo = new InfiniumGTCFile(new DataInputStream(new FileInputStream(INPUT.get(1))), infiniumNormalizationManifest);
+            compareGTCFiles(gtcFileOne, gtcFileTwo);
+
+            //report errors and exit 1 if any are detected.
+            if (!errors.isEmpty()) {
+                for (String error : errors) {
+                    log.error(error);
+                }
+                return 1;
+            }
+        } catch (IOException | IllegalAccessException | InvocationTargetException e) {
+            throw new PicardException("File error: ", e);
+        }
+        return 0;
+    }
+
+    private void compareGTCFiles(InfiniumGTCFile gtcFileOne, InfiniumGTCFile gtcFileTwo) throws InvocationTargetException, IllegalAccessException {
+        //compare all fields we expect won't change.
+        Method[] methods = gtcFileOne.getClass().getMethods();
+        for (Method method : methods) {
+            //skip ignored methods.
+            if (IGNORED_METHODS.contains(method.getName())) {
+                continue;
+            }
+            //compare all getters
+            if (method.getName().startsWith("get") && method.getGenericParameterTypes().length == 0) {
+                //if we have a version and they don't match we just want a warning
+                //if getter returns an array compare all array values otherwise do an Object compare.
+                //if getter returns an array of arrays do deep compare
+                if (method.getName().equals("getFileVersion")) {
+                    compareVersions(method.invoke(gtcFileOne), method.invoke(gtcFileTwo));
+                } else if (method.getReturnType().isArray() && method.getReturnType().getComponentType().isArray()) {
+                    compareArrayOfArrays(method.invoke(gtcFileOne), method.invoke(gtcFileTwo),
+                            method.getName());
+                } else if (method.getReturnType().isArray()) {
+                    compareArrays(method.invoke(gtcFileOne), method.invoke(gtcFileTwo),
+                            method.getName());
+                } else {
+                    compare(method.invoke(gtcFileOne),
+                            method.invoke(gtcFileTwo), method.getName());
+                }
+            }
+        }
+    }
+
+    private void compareVersions(Object versionOne, Object versionTwo) {
+        if (!versionOne.equals(versionTwo)) {
+            log.warn(String.format("File versions do not match ( %s vs %s )",
+                    versionOne, versionTwo));
+        }
+    }
+
+    private void compare(Object objectOne, Object objectTwo, String type) {
```

Comment:
It's little confusing that `type` is used here when this is actually the method name.

Reply:
renamed it.